### ISSUE TITLE
Mock github in tc-github via nock

### ISSUE THREAD

**Issue by djmitche** [taskcluster/taskcluster#3985]
Mocking the Octokit interface is hard to keep in sync with Octokit, and is getting harder as they move more functionality to plugins.  It also means we're not actually testing our use of those plugins.

Let's switch to mocking GitHub at the HTTP level.  That API is better (?) defined and documented, and of necessity more stable.

In particular, that will let us use the Octokit pagination support, which tbh we should be using for all `*.list*` calls.

**Comment by escapewindow** [comment]
@sladyn98 Optional, potentially related.

**Comment by sladyn98** [comment]
Yes definitely, I can pick this up, haven't done this before though, could give it a shot, what would a be a good starting point, I read a bit of the docs for nock, so which tests would we like to get started with and any specific gotchas to keep  in mind

**Comment by djmitche** [comment]
There's a bunch of code to "fake" a GitHub client in `services/github/test/github-auth.js`.  I think we'd like to replace all of that with nocks that act like github's HTTP API.

This will mean building a new bit of test harness that supports setup / teardown as well as all of the fake-related methods like `setRepositories` and `getComments`.  I'd suggest finding a test that only requires one or two GitHub API calls, and initially working to get that test passing.  Remove the existing fakes, and start building out your test harness just implementing the few required HTTP URLs.  Once you've got that working, we can have an early look at the PR, and then you can go on to implement all of the other methods and get the whole test suite passing.

**Comment by sladyn98** [comment]
@djmitche Sure I will give this a shot and see how far I can get and raise a draft PR
